pywbem_mock/_instancewriteprovider.py

```diff
@@ -270,192 +213,231 @@ def __init__(self, cimrepository=None):
+        # not exist yet.
         try:
-            instance_store.create(new_instance.path, new_instance)
+            instance_store.create(NewInstance.path, NewInstance)
         except ValueError:
             raise CIMError(
                 CIM_ERR_ALREADY_EXISTS,
-                _format("NewInstance {0!A} already exists in namespace "
-                        "{1!A}.", new_instance.path, namespace))
+                _format("New instance {0!A} already exists in namespace "
+                        "{1!A}.", NewInstance.path, namespace))
 
-        # Create instance returns model path
-        return new_instance.path
+        # CreateInstance returns the instance path of the new instance
+        return NewInstance.path
 
     ####################################################################
     #
-    #   ModifyInstance server request
+    #   ModifyInstance mock WBEM server responder
     #
     ####################################################################
 
     def ModifyInstance(self, ModifiedInstance,
                        IncludeQualifiers=None, PropertyList=None):
         # pylint: disable=invalid-name,line-too-long,unused-argument
         """
-        Implements a WBEM server responder for
+        Implements a mock WBEM server responder for
         :meth:`pywbem.WBEMConnection.CreateInstance`.
 
-        Modify a CIM instance in the CIM repository.
-
-        NOTE: This method includes namespace within the path element
-        of the ModifiedInstance rather than as a separate input parameter.
+        Modify an existing CIM instance in the CIM repository of the mock WBEM
+        server.
+
+        NOTE: This method specifies the namespace in ModifiedInstance.path
+        rather than as a separate input parameter.
+
+        The modification of the instance in the CIM repository that is performed
+        by the provider should be limited to property value changes (including
+        addition of properties if not yet set on the instance), because
+        instance-level qualifiers have been deprecated in CIM.
+
+        The set of properties that are to be modified on the CIM instance, is:
+
+        - If PropertyList is not None: All properties specified in the
+          PropertyList parameter, to the value specified in ModifiedInstance
+          (inlcuding `None` for setting to NULL).
+
+          If such a property is not specified in ModifiedInstance, it must be
+          set to the default value of the property exposed by the creation
+          class of the instance in the CIM repository, or to any other value
+          (inlcuding `None`) if the property does not define a default value.
+
+        - If PropertyList is None: All properties specified in the
+          ModifiedInstance parameter, to the values specified there (inlcuding
+          `None` for setting to NULL).
+
+        Key properties must not be modified. Depending on the provider,
```

Comment:
Change "Depending on the provider" to "A user-defined provider"

Reply:
DONE - Changed to "Depending on the model implemented,"